server/schedule/operator/builder.go

```diff
@@ -279,7 +279,9 @@ func (b *Builder) buildSteps(kind OpKind) (OpKind, error) {
 		}
 		if plan.add != nil {
 			b.execAddPeer(plan.add)
-			kind |= OpRegion
+			if kind&OpMerge == 0 {
```

Comment:
Can we change the condition of the `AllowBalance` rather than change this?
Merge may move peers or only transfer leaders.